inc/zoo/AnyCallable.h

```diff
@@ -65,8 +66,60 @@ struct AnyCallable<TypeErasureProvider, R(Args...)>: TypeErasureProvider {
                 const_cast<AnyCallable &>(*this)
             );
     }
+
+    void swap(AnyCallable& other) noexcept {
+        TypeErasureProvider::swap(other);
+        std::swap(targetInvoker_, other.targetInvoker_);
+    }
+
+    explicit operator bool() const noexcept {
+        return !empty();
+    }
+
+    const std::type_info& target_type() const noexcept {
+        return empty() ? typeid(void) : this->type();
+    }
+
+    bool empty() const noexcept {
+        return targetInvoker_ == emptyInvoker_;
+    }
+
+    template< class T >
+    T* target() noexcept {
+        using uncvr_t = std::remove_cv_t<std::remove_reference_t<T>>;
```

Comment:
`std::decay` does this. Also, I tend to use first-letter upper case camel case for types, except in the cases the components, such as `zoo::function` are so fundamental or similar to the libraries in the official standard that I use the conventions of the standard, but for non-user-facing code, I prefer using my normal capitalization